### PR TITLE
[Mod] Pluralize properly in userinfo

### LINKED ISSUE #4379: [serverinfo/userinfo] Pluralize strings
# Commands improvement

#### Command name

`[p]userinfo` and `[p]serverinfo`

#### What cog is this command from?

From `Mod` cog.

#### What improvement are you expecting to?

I would like to see all strings pluralized (ie. `Text Channels` being `Text Channel` if there's only one channel and so on).

#### What actually happened?

Roles and previous nickname/names aren't pluralized.

### PR DESCRIPTION
**Pull request by Dav-Git**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Resolves #4379

### CODE REVIEW

**Comment by Dav-Git** [comment]
I didn't edit anything in serverinfo because serverinfo gramatically makes sense with he plural

**Review by madebylydia — DISMISSED** [review]

**Comment by TurnrDev** [comment]
I really think it would be useful if the translation module supported [Plural-Forms](https://www.gnu.org/software/gettext/manual/html_node/Translating-plural-forms.html).

**Review by TurnrDev — DISMISSED** [review]

**Comment by Dav-Git** [comment]
Yeah, I tried to make this as translation safe as possible by using to strings as not all languages pluralize by appending an s.

File: redbot/cogs/mod/names.py [hunk 1]
```diff
@@ -245,15 +245,25 @@ async def userinfo(self, ctx, *, user: discord.Member = None):
...
-            data.add_field(name=_("Roles"), value=role_str, inline=False)
+            data.add_field(
+                name=_("Roles") if len(roles) > 2 else _("Role"), value=role_str, inline=False
+            )  # len(roles) > 2 because @everyone is a "role" which is just not displayed.
         if names:
             # May need sanitizing later, but mentions do not ping in embeds currently
             val = filter_invites(", ".join(names))
-            data.add_field(name=_("Previous Names"), value=val, inline=False)
+            data.add_field(
+                name=_("Previous Names") if len(val) > 1 else _("Previous Name"),
```
Comment (Jackenmen) [inline redbot/cogs/mod/names.py]: You need to check the length of names list, not the length of the resulting string.
```suggestion
                name=_("Previous Names") if len(names) > 1 else _("Previous Name"),
```

File: redbot/cogs/mod/names.py [hunk 2]
```diff
@@ -245,15 +245,25 @@ async def userinfo(self, ctx, *, user: discord.Member = None):
...
+                value=val,
+                inline=False,
+            )
         if nicks:
             # May need sanitizing later, but mentions do not ping in embeds currently
             val = filter_invites(", ".join(nicks))
-            data.add_field(name=_("Previous Nicknames"), value=val, inline=False)
+            data.add_field(
+                name=_("Previous Nicknames") if len(val) > 1 else _("Previous Nickname"),
```
Comment (Jackenmen) [inline redbot/cogs/mod/names.py]: You need to check the length of names list, not the length of the resulting string.
```suggestion
                name=_("Previous Nicknames") if len(nicks) > 1 else _("Previous Nickname"),
```

**Review by Jackenmen — CHANGES_REQUESTED** [review]

**Comment by Dav-Git** [comment]
lmao how did this derp sneak through? Thanks for noticing :)

File: redbot/cogs/mod/names.py [hunk 3]
```diff
@@ -245,15 +245,25 @@ async def userinfo(self, ctx, *, user: discord.Member = None):
         data.add_field(name=_("Joined Discord on"), value=created_on)
         data.add_field(name=_("Joined this server on"), value=joined_on)
         if role_str is not None:
-            data.add_field(name=_("Roles"), value=role_str, inline=False)
+            data.add_field(
+                name=_("Roles") if len(roles) > 2 else _("Role"), value=role_str, inline=False
+            )  # len(roles) > 2 because @everyone is a "role" which is just not displayed.
```
Comment (Jackenmen) [inline redbot/cogs/mod/names.py]: Sorry, I somehow missed this. We already strip the `@everyone` role above (see line 173), so this results in incorrect results.
```suggestion
                name=_("Roles") if len(roles) > 1 else _("Role"), value=role_str, inline=False
            )
```

**Review by Jackenmen — CHANGES_REQUESTED** [review]

**Review by Jackenmen — APPROVED** [review]
LGTM, thanks!

**Pull request merged**